portal/server/helpers/pageSpecificValidationErrors.js

```diff
@@ -4,126 +4,115 @@
 // - - the entities 'Check your answers' page (AKA 'Preview' page) has been viewed
 // - - one of entities forms has had a value submitted
 // by having pageSpecificValidationErrors logic in the UI, we remain decoupled = require(the API).
-
+const cloneDeep = require('lodash/cloneDeep')
 const errorHref = require('./errorHref');
 const generateErrorSummary = require('./generateErrorSummary');
-const {
-  requiredFieldsArray,
-  filterErrorList,
-} = require('./pageFields');
+const { requiredFieldsArray, filterErrorList } = require('./pageFields');
 
 const allFieldsArray = (fields) => {
-  const { OPTIONAL_FIELDS } = fields;
-  const allFields = requiredFieldsArray(fields);
+ const { OPTIONAL_FIELDS } = fields;
+ const allFields = requiredFieldsArray(fields);
 
-  if (OPTIONAL_FIELDS) {
-    allFields.push(...OPTIONAL_FIELDS);
-  }
+ if (OPTIONAL_FIELDS) {
+   allFields.push(...OPTIONAL_FIELDS);
+ }
 
-  return allFields;
+ return allFields;
 };
 
 const shouldReturnRequiredValidation = (fields, fieldValues) => {
-  const allFields = allFieldsArray(fields);
-
-  if (fieldValues.viewedPreviewPage) {
-    return true;
-  }
-
-  const totalFieldValues = Object.keys(fieldValues).filter((fieldName) =>
-    allFields.includes(fieldName)
-    && fieldValues[fieldName] !== null
-    && fieldValues[fieldName].length > 0);
-
-  if (totalFieldValues.length > 0) {
-    return true;
-  }
-
-  return false;
+ const allFields = allFieldsArray(fields);
+
+ if (fieldValues.viewedPreviewPage) {
+   return true;
+ }
+ const totalFieldValues = Object.keys(fieldValues).filter(
+   (fieldName) =>allFields.includes(fieldName)
+ );
+ if (totalFieldValues.length > 0) {
+   return true;
+ }
+
+ return false;
 };
```

Comment:
I feel removing: 

```
    && fieldValues[fieldName] !== null
    && fieldValues[fieldName].length > 0
```

is interesting (although I can see why this fixes things) -- are we sure it has no unexpected consiquences?